pypesto/visualize/optimize.py

```diff
@@ -0,0 +1,70 @@
+import matplotlib.pyplot as plt
+import matplotlib.colors as colors
+import matplotlib.cm as cm
+from matplotlib.ticker import MaxNLocator
+from scipy import cluster
+import numpy as np
+
+
+def waterfall(result, ax=None):
+    """
+    Plot waterfall plot.
+
+    Parameters
+    ----------
+
+    result: pypesto.Result
+        Optimization result obtained by 'optimize.py'
+
+    ax: matplotlib.Axes, optional
+        Axes object to use.
+
+    Returns
+    -------
+
+    ax: matplotlib.Axes
+        The plot axes.
+    """
+
+    # axes
+    if ax is None:
+        fig, ax = plt.subplots()
+
+    # extract cost function values from result
+    result_fval = result.optimize_result.get_for_key('fval')
+    result_fval = np.reshape(result_fval, [len(result_fval), 1])
+    startpoints = range(1, len(result_fval) + 1)
+
+    # cluster
+    clust = cluster.hierarchy.fcluster(
+        cluster.hierarchy.linkage(result_fval),
+        0.1, criterion='distance')
+    uclust, ind_clust = np.unique(clust, return_inverse=True)
+    clustsize = np.zeros(len(uclust))
+    for iclustsize in range(len(uclust)):
+        clustsize[iclustsize] = sum(clust == uclust[iclustsize])
+
+    # assign colors
+    jet = plt.get_cmap('jet')
```

Comment:
Don't select color maps, keep styling at the bare minimum. The user should take care of that in a centralized way. We can provide a default style though (https://matplotlib.org/users/customizing.html).